huaweicloud/services/swr/data_source_huaweicloud_swr_shared_repositories.go

```diff
@@ -0,0 +1,236 @@
+		return diag.Errorf("error creating SWR client: %s", err)
+	}
+
+	// listSharedRepositories: Query the list of SWR shared repositories.
+	resp, err := getSharedRepositories(client, d)
+	if err != nil {
+		return diag.Errorf("error retrieving SWR shared repositories: %s", err)
+	}
+
+	randUUID, err := uuid.GenerateUUID()
+	if err != nil {
+		return diag.Errorf("unable to generate ID: %s", err)
+	}
+	d.SetId(randUUID)
+
+	mErr := multierror.Append(
+		nil,
+		d.Set("region", region),
+		d.Set("repositories", flattenSharedRepositoriesResponse(resp, d)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func getSharedRepositories(client *golangsdk.ServiceClient, d *schema.ResourceData) ([]interface{}, error) {
+	listSharedRepositoriesHttpUrl := "v2/manage/shared-repositories"
+	listSharedRepositoriesPath := client.Endpoint + listSharedRepositoriesHttpUrl
+	listSharedRepositoriesQueryParams := buildListSharedRepositoriesQueryParams(d)
+	listSharedRepositoriesPath += listSharedRepositoriesQueryParams
+
+	listSharedRepositoriesOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders:      map[string]string{"Content-Type": "application/json"},
+	}
+
+	results := make([]interface{}, 0)
+	offset := 0
+	for {
+		currentPath := fmt.Sprintf("%slimit=10&offset=%d", listSharedRepositoriesPath, offset)
+		listSharedRepositoriesResp, err := client.Request("GET", currentPath, &listSharedRepositoriesOpt)
+		if err != nil {
+			return nil, fmt.Errorf("error querying SWR shared repositories: %s", err)
+		}
+		listSharedRepositoriesRespBody, err := utils.FlattenResponse(listSharedRepositoriesResp)
+		if err != nil {
+			return nil, fmt.Errorf("error retrieving SWR shared repositories: %s", err)
+		}
+		sharedRepositories := listSharedRepositoriesRespBody.([]interface{})
+		total := 0
+		if len(sharedRepositories) > 0 {
+			total = int(utils.PathSearch("total_range", sharedRepositories[0], float64(0)).(float64))
+		}
+		results = append(results, sharedRepositories...)
+		offset += len(sharedRepositories)
+		if offset == total {
+			break
+		}
+	}
+
+	return results, nil
+}
+
+func buildListSharedRepositoriesQueryParams(d *schema.ResourceData) string {
+	res := ""
+
+	if v, ok := d.GetOk("organization"); ok {
+		res = fmt.Sprintf("%s&namespace=%v", res, v)
+	}
+	if v, ok := d.GetOk("name"); ok {
+		res = fmt.Sprintf("%s&name=%v", res, v)
+	}
+	if v, ok := d.GetOk("center"); ok {
+		res = fmt.Sprintf("%s&center=%v", res, v)
+	}
+	if v, ok := d.GetOk("domain_name"); ok {
+		res = fmt.Sprintf("%s&domain_name=%v", res, v)
+	}
+
+	if res != "" {
+		return fmt.Sprintf("?%s&", res[1:])
+	}
+	return "?"
+}
+
+func flattenSharedRepositoriesResponse(rawParams []interface{}, d *schema.ResourceData) []interface{} {
+	if len(rawParams) == 0 {
+		return nil
+	}
+	results := make([]interface{}, 0)
+	for _, v := range rawParams {
+		name := utils.PathSearch("name", v, "").(string)
+		if val, ok := d.GetOk("name"); ok && name != val {
+			continue
+		}
```

Comment:
the query params have contains name, why use the name to filter the result angin?